### PR TITLE
fix: Internal error message on login with missing credential

### LINKED ISSUE #2367: Logging in without a password throws syntax error
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and [Parse Dashboard](https://github.com/parse-community/parse-dashboard/releases). 

### Issue Description
Screenshot 2023-01-25 at 1 57 30 pm

### Environment

Dashboard
- Parse Dashboard version: `FILL_THIS_OUT`
- Browser (Safari, Chrome, Firefox, Edge, etc.): `FILL_THIS_OUT`
- Browser version: `FILL_THIS_OUT`

Server
- Parse Server version: `FILL_THIS_OUT`
- Operating system: `FILL_THIS_OUT`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `FILL_THIS_OUT`

Database
- System (MongoDB or Postgres): `FILL_THIS_OUT`
- Database version: `FILL_THIS_OUT`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description

Passport can throw error strings, which cannot be converted to JSON. In this case, the errors should not be overridden.

Closes: #2367

### TODOs before merging

- [x] Add tests

### CODE REVIEW

File: src/login/Login.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default class Login extends React.Component {
         this.errors = json.text
         otpLength = json.otpLength;
       } catch (e) {
-        this.errors = `could not pass error json: ${e}`;
+        /* */
```
Comment (mtrezza) [inline src/login/Login.js]: Shouldn't this set at least a generic error message? What is `this.errors` used for?

Comment (dblythy) [reply src/login/Login.js]: this.errors is errors passed from passport. They can be a string "invalid credentials" or a JSON string with an error

Comment (mtrezza) [reply src/login/Login.js]: I'm not comfortable with an empty catch block - lint should complain here. In which cases could an error be thrown and wouldn't we want to show a specific error message in that case?

Comment (dblythy) [reply src/login/Login.js]: I’ll rethink this

Comment (mtrezza) [reply src/login/Login.js]: If it makes sense maybe something like:

```suggestion
        this.errors = this.errors ?? `Error: ${JSON.stringify(e)}`;
```

Other than that the PR seems to fix the issue:
image

**Review by mtrezza — APPROVED** [review]
Looks good, tested in preview.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0-alpha.8](https://github.com/parse-community/parse-dashboard/releases/tag/5.1.0-alpha.8)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0-beta.2](https://github.com/parse-community/parse-dashboard/releases/tag/5.1.0-beta.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0](https://github.com/parse-community/parse-dashboard/releases/tag/5.1.0)

**Pull request merged**